malloc_count.c

```diff
@@ -122,6 +125,30 @@ extern void malloc_count_reset_peak(void)
     peak = curr;
 }
 
+/* user function to return total allocation volume */
+extern size_t malloc_count_total(void)
+{
+    return local_total;
+}
+
+/* user function to reset total allocation volume to 0 */
+extern void malloc_count_reset_total(void)
+{
+    local_total = 0;
```

Comment:
This local_total counter is bad: it introduces global state.
The local counter can also be done in a much safer way, by saving the current alloc twice and calculating the delta.